E2e/feature/ri 6231 spike] fix the rounding issue for json data type

KIvanow opened the pull request:
Tested it with bigints as numbers, bigints as strings both as the only element in the json and the same two options as part of arrays and objects.

In the added changes testes were performed with very large jsons that trigger partial retrievals of the information (kudos to @romansergeenkosofteq for pointing the different case there). For both cases we are not parsing the data on the BE before sending it back to the UI (to avoid the bigint issues). All of the parsing is done afterwards in the client. Separated the parsing function into 2 different ones
1 - parsing when the data is simple and whole (we've retrieved the entire json and simply need to parse it)
2 - parsing the partially retrieved data. In that case the data is not just a string or a JSON valid object but looks more like this:
`
"data": [
        {
            "key": "_id",
            "path": "[0][\"_id\"]",
            "cardinality": 1,
            "type": "string",
            "value": "\"6735edc1021e565aca45f9fe\""
        },
        {
            "key": "index",
            "path": "[0][\"index\"]",
            "cardinality": 1,
            "type": "integer",
            "value": "1188950299261208942"
        },
`
So separating the two seemed appropriate.

Tests were added for both parsing cases.

romansergeenkosofteq reviewed (commented):
could you please add tests
also, could you please test the case when we get not the whole JSON (when it is big)

KIvanow commented:
> could you please add tests also, could you please test the case when we get not the whole JSON (when it is big)

@romansergeenkosofteq do we have a big json anywhere is test files or docs that we can use for the second thing?

romansergeenkosofteq reviewed (commented):
1. I am not sure that it is a good solution to parse JSON in the slice, I would recommend set in the store what we receive from BE. The responsibility how to work with that data should be on component layer.
2. Please, use our linter (we don't use semicolons) and add empty line at the end of the files

KIvanow commented:
> 1. I am not sure that it is a good solution to parse JSON in the slice, I would recommend set in the store what we receive from BE. The responsibility how to work with that data should be on component layer.

Updated the approach to not parse the jsons in the slice, but in a component directly. Added it in a util in case 
we need it in multiple other places down the line.


> 2. Please, use our linter (we don't use semicolons) and add empty line at the end of the files
Thank you for the tip!

@ArtemHoruzhenko I've also updated the BE tests to check the new structure. Surprisingly the DTO didn't require changes

ArtemHoruzhenko reviewed (dismissed):
lgtm

romansergeenkosofteq reviewed (commented):
@KIvanow there are some ESLint errors in the utils.ts (@typescript-eslint/indent, no-trailing-spaces, arrow-parens, no-param-reassign, import/order). Please use our eslint settings for IDE.

KIvanow commented on redisinsight/ui/src/pages/browser/modules/key-details/components/rejson-details/utils/utils.ts:
```diff
@@ -61,3 +62,72 @@ export const getBrackets = (type: string, position: 'start' | 'end' = 'start') =
+        default:
+          return value
+      }
+    }
```
I'd keep it as is for now. The function is relatively small/short and we don't have any specific use case in the changes to specifically parse just by type. We can always separate it whenever we need it down the line

romansergeenkosofteq reviewed (approved):
lgtm

KIvanow commented:
Testing is done :)

romansergeenkosofteq commented on redisinsight/ui/src/pages/browser/modules/key-details/components/rejson-details/rejson-details/RejsonDetails.tsx:
```diff
@@ -32,8 +32,14 @@ const RejsonDetails = (props: BaseProps) => {
+        data: parseJsonData(data?.data)
+      }))
+      .catch(reject)
+  })
```
I would suggest parse json in the fetchVisualisationResults

romansergeenkosofteq replied:
`fetchVisualisationResults` do not store results. It's ok to transform result of BE and return prepared data.

KIvanow replied:
I see what you mean. Updated it.

KIvanow commented on redisinsight/ui/src/pages/browser/modules/key-details/components/rejson-details/rejson-scalar/RejsonScalar.tsx:
```diff
@@ -36,7 +36,15 @@ const RejsonScalar = (props: JSONScalarProps) => {
+      setChangedValue('null')
+    } else {
+      setChangedValue(value)
+    }
```
Do you mean like so -> https://github.com/RedisInsight/RedisInsight/pull/4089/commits/238acd98a5a502e1fb4ba3dae4c64554b96d2517 having a function wrapping up this functionality, instead of doing it directly in the useEffect, or something else?

romansergeenkosofteq replied:
yes, but could you please move it to utils and call smth like stringifyScalarValue?

KIvanow replied:
done, covered it with tests as well https://github.com/RedisInsight/RedisInsight/pull/4089/commits/2d08a99edf1991533d0a28cc3516f3e81c473225

I usually wait until something needs the function from a different component,before separating it in a util, but I can see the benefit of the other approach as well

romansergeenkosofteq reviewed (commented):
please, do lint changed FE files, there are some lint errors

KIvanow commented:
> please, do lint changed FE files, there are some lint errors

@romansergeenkosofteq done and added the linter to be triggered on save for all of my devices. Sorry for the recurring issue
